src/service/kafkaRelayPlugin.ts

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: Apache-2.0
+import { Kafka, logLevel, type Producer } from 'kafkajs';
+import { additionalEnvironmentVariables, type Configuration } from '../config';
+import type { LoggerService } from '@tazama-lf/frms-coe-lib';
+import type { Apm } from '@tazama-lf/frms-coe-lib/lib/services/apm';
+import { validateProcessorConfig } from '@tazama-lf/frms-coe-lib/lib/config/processor.config';
+import * as fs from 'fs';
+import type { ITransportPlugin } from '@tazama-lf/frms-coe-lib/lib/interfaces/relay-service/ITransportPlugin';
+
+export default class KafkaRelayPlugin implements ITransportPlugin {
+  private readonly kafka: Kafka;
+  private producer?: Producer;
+  private loggerService?: LoggerService;
+  private apm?: Apm;
+  private readonly configuration: Configuration;
+  private readonly maxInFlight: number;
```

Comment:
- Remove maxInFlight (already defined in configuration property)
- Make producer non optional and readonly (create producer in constructor: see example)

```suggestion
export default class KafkaRelayPlugin implements ITransportPlugin {
  private apm?: Apm;
  private loggerService?: LoggerService;
  private readonly kafka: Kafka;
  private readonly producer: Producer;
  private readonly configuration: Configuration;
```